### PR TITLE
Support updated matching syntax for inhibit rules in AlertManagerConfig CRD

### LINKED ISSUE #4087: Support target_matchers and source_matchers syntax (Alertmanager >= v0.22.0) in AlertmanagerConfig CRD
**What happened?**
Hello, I'm trying to set some inhibit_rules using the new syntax in 0.22.0:
https://prometheus.io/docs/alerting/latest/configuration/#inhibit_rule
where `target_match` , `source_match` have been deprecated and replaced by:
`target_matchers` and `source_matchers` that now support PromQL syntax for matching.

However the AlertmanagerConfig don't support them:

```
❯ kubectl explain alertmanagerconfig.spec.inhibitRules
KIND:     AlertmanagerConfig
VERSION:  monitoring.coreos.com/v1alpha1

RESOURCE: inhibitRules <[]Object>

DESCRIPTION:
     List of inhibition rules. The rules will only apply to alerts matching the
     resource’s namespace.

     InhibitRule defines an inhibition rule that allows to mute alerts when
     other alerts are already firing. See
     https://prometheus.io/docs/alerting/latest/configuration/#inhibit_rule

FIELDS:
   equal	<[]string>
     Labels that must have an equal value in the source and target alert for the
     inhibition to take effect.

   sourceMatch	<[]Object>
     Matchers for which one or more alerts have to exist for the inhibition to
     take effect. The operator enforces that the alert matches the resource’s
     namespace.

   targetMatch	<[]Object>
     Matchers that have to be fulfilled in the alerts to be muted. The operator
     enforces that the alert matches the resource’s namespace.

```

So the config changes are ignored..

**How to reproduce it (as minimally and precisely as possible)**:
Deploy Prom-operator in v0.48.0, use a simple Alertmanager config that use the new matchers such as:
https://github.com/prometheus/alertmanager/blob/master/doc/examples/simple.yml

* Prometheus Operator version: 
v0.48.0

Thanks

### PR DESCRIPTION
**Pull request by PhilipGough**
## Description

Fixes #4087 

## Type of change

_What type of changes does your code introduce to the Prometheus operator? Put an `x` in the box that apply._

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [x] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [ ] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Changelog entry

_Please put a one-line changelog entry below. This will be copied to the changelog file during the release process._

```release-note
Support updated matching syntax for inhibit rules in AlertManagerConfig CRD
```

### CODE REVIEW

File: pkg/alertmanager/amcfg.go [hunk 1]
```diff
@@ -711,54 +713,80 @@ func (cg *configGenerator) convertPushoverConfig(ctx context.Context, in monitor
 	return out, nil
 }
 
-func convertInhibitRule(in *monitoringv1alpha1.InhibitRule, crKey types.NamespacedName) *inhibitRule {
+func (cg *configGenerator) convertInhibitRule(in *monitoringv1alpha1.InhibitRule, crKey types.NamespacedName) *inhibitRule {
```
[the remark this answers is not in the document]
Comment (PhilipGough) [reply pkg/alertmanager/amcfg.go:716]: Going with `2.` after discussion on slack

File: Documentation/api.md [hunk 2]
```diff
@@ -1405,7 +1407,8 @@ Matcher defines how to match on alert's labels.
 | ----- | ----------- | ------ | -------- |
 | name | Label to match. | string | true |
 | value | Label value to match. | string | true |
-| regex | Whether to match on equality (false) or regular-expression (true). | bool | false |
+| matchType | Match operation available with AlertManager >= v0.22.0 and takes precedence over Regex (deprecated) if non-empty. | MatchType | false |
```
[the remark this answers is not in the document]
Comment (PhilipGough) [reply Documentation/api.md:1408]: This is not going to generate. Opened https://github.com/prometheus-operator/prometheus-operator/issues/4376 to track.

[the remark this answers is not in the document]
Comment (simonpasquier) [reply Documentation/api.md:1409]: thanks! resolving this for now.

**Review by simonpasquier — APPROVED** [review]
lgtm

**Pull request merged**